internal/matrix/matrix_test.go

```diff
@@ -0,0 +1,51 @@
+package matrix
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestGenerateFibMatrix(t *testing.T) {
+
+	data := []struct {
+		name     string
+		rows     int
+		columns  int
+		expected [][]int64
+		err      error
```

Comment:
Changing this to be type bool should make it more readable, but it tests pass like this then it should be fine.